File: MockInterview.Frontend/src/types/includeOptions.ts
import {IMappable} from "./common";

export interface IIncludeOptions  {
    includeModels: Array<string>;
}

export class IncludeOptions implements IMappable<IIncludeOptions> {
    includeModels!: Array<string>;

    mapFrom(data: IIncludeOptions) {
        this.includeModels = data.includeModels;
    }
}

File: MockInterview.Backend/MockInterview.DAL/Models/Query/IncludeOptions.cs
using MockInterview.Core.Models.Common;

namespace MockInterview.DAL.Models.Query;

/// <summary>
/// Represents including options
/// </summary>
/// <typeparam name="TEntity"></typeparam>
public class IncludeOptions<TEntity> where TEntity : class, IEntity
{
    public IncludeOptions()
    {
        IncludeModels = new List<string>();
    }

    public IEnumerable<string> IncludeModels { get; set; }
}